subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_adv_firmware.c

```diff
@@ -378,9 +378,6 @@ static int update_arguments_parse(uint16_t obj_inst_id, char *args, uint16_t arg
 			if (s_ptr && e_ptr) {
 				args_len -= e_ptr - args;
 				temp = e_ptr + 1;
-				/* Mark end prefix by end of string */
-				*e_ptr = '\0';
```

Comment:
@rlubos @SeppoTakalo with the recent discord reported issue of non-null terminated strings, would this be an issue?

Reply:
I can test this by unit test

Reply:
I think the question is valid. Why do we remove the `\0` marking? Did it cause some side effects?

I know that the end result of this string is fed into `strtoul()` which will end the parsing at the `>` character that we have matched earlier. However, I still think it would be good idiom to check that `*endptr` from the `strtoul()` points to `\0` so we know we have converted all characters.

Or maybe we could check there that we end at the `>`